src/libData/AccountData/AccountStoreTrie.tpp

```diff
@@ -0,0 +1,130 @@
+* GPLv3.0 are those programs that are located in the folders src/depends and tests/depends 
+* and which include a reference to GPLv3 in their program files.
+**/
+
+#include "libPersistence/ContractStorage.h"
+
+template<class DB, class MAP>
+AccountStoreTrie<DB, MAP>::AccountStoreTrie()
+    : m_db(is_same<DB, OverlayDB>::value ? "state" : "")
+{
+    m_state = dev::SpecificTrieDB<dev::GenericTrieDB<DB>, Address>(&m_db);
+}
+
+template<class DB, class MAP> void AccountStoreTrie<DB, MAP>::Init()
+{
+	AccountStoreSC<MAP>::Init();
+    m_state.init();
+    prevRoot = m_state.root();
+}
+
+template<class DB, class MAP>
+Account* AccountStoreTrie<DB, MAP>::GetAccount(const Address& address)
+{
+    Account* account = AccountStoreBase<MAP>::GetAccount(address);
+    if (account != nullptr)
+    {
+        return account;
+    }
+
+    string accountDataString = m_state.at(address);
+    if (accountDataString.empty())
+    {
+        return nullptr;
+    }
+
+    dev::RLP accountDataRLP(accountDataString);
+    if (accountDataRLP.itemCount() != 4)
+    {
+        LOG_GENERAL(WARNING, "Account data corrupted");
+        return nullptr;
+    }
+
+    auto it2 = this->m_addressToAccount->emplace(
+        std::piecewise_construct, std::forward_as_tuple(address),
+        std::forward_as_tuple(accountDataRLP[0].toInt<uint256_t>(),
+                              accountDataRLP[1].toInt<uint256_t>()));
+
+    // Code Hash
+    if (accountDataRLP[3].toHash<h256>() != h256())
+    {
+        // Extract Code Content
+        it2.first->second.SetCode(
+            ContractStorage::GetContractStorage().GetContractCode(address));
+        if (accountDataRLP[3].toHash<h256>() != it2.first->second.GetCodeHash())
+        {
+            LOG_GENERAL(WARNING, "Account Code Content doesn't match Code Hash")
+            this->m_addressToAccount->erase(it2.first);
+            return nullptr;
+        }
+        // Storage Root
+        it2.first->second.SetStorageRoot(accountDataRLP[2].toHash<h256>());
+    }
+
+    return &it2.first->second;
+}
+
+template<class DB, class MAP>
+bool AccountStoreTrie<DB, MAP>::UpdateStateTrie(const Address& address,
+                                           const Account& account)
+{
+    //LOG_MARKER();
+
+    dev::RLPStream rlpStream(4);
+    rlpStream << account.GetBalance() << account.GetNonce()
+              << account.GetStorageRoot() << account.GetCodeHash();
+    m_state.insert(address, &rlpStream.out());
+
+    return true;
+}
+
+template<class DB, class MAP> 
+h256 AccountStoreTrie<DB, MAP>::GetStateRootHash() const
+{
+    LOG_MARKER();
+
+    return m_state.root();
+}
+
+template<class DB, class MAP> 
+bool AccountStoreTrie<DB, MAP>::UpdateStateTrieAll()
+{
+    bool ret = true;
+    for (auto entry : *(this->m_addressToAccount))
+    {
+        if (!UpdateStateTrie(entry.first, entry.second))
+        {
+            ret = false;
+            break;
```

Comment:
Just return false inside the loop and return true in the end of this function.
Needless to use "ret" variable.